sdk/python/touca/_transport.py

```diff
@@ -11,27 +11,11 @@
 
 class Transport:
     def __init__(self):
-        self._token = None
         self._pool = PoolManager(cert_reqs="CERT_REQUIRED", ca_certs=certifi.where())
 
-    def authenticate(self, api_url: str, api_key: str):
-        from touca._options import ToucaError
-
-        if self._token and self._api_key == api_key and self._api_url == api_url:
-            return
-        self._api_key = api_key
+    def configure(self, api_url: str, api_key: str):
         self._api_url = api_url
-        response = self.request(
-            method="POST",
-            path="/client/signin",
-            body={"key": api_key},
-        )
```

Comment:
Since we used to call `authenticate` as part of configuring the test runner or the lower-level core library, and it functioned as a way to also validate the API Key and API URL and make sure they are correct, I'd vote that we continue to make a post request here.

Reply:
If the endpoint we're calling remains as `client/signin`, perhaps we should modify it to return 204 if API Key header is set. For backward compatibility, we cannot add `isCilentAuthenticated` middleware to it, yet.